mmv1/products/dialogflowcx/api.yaml

```
@@ -0,0 +1,115 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: DialogflowCX
+display_name: Dialogflow CX
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://{{location}}-dialogflow.googleapis.com/v3/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Dialogflow API
+    url: https://console.cloud.google.com/apis/library/dialogflow.googleapis.com
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Agent'
+    base_url: "projects/{{project}}/locations/{{location}}/agents"
+    self_link: "{{name}}"
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      Agents are best described as Natural Language Understanding (NLU) modules that transform user requests into actionable data. You can include agents in your app, product, or service to determine user intent and respond to the user in a natural way.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/dialogflow/cx/docs'
+      api: 'https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        output: true
+        description: |
+          The unique identifier of the agent. Required for the Agents.UpdateAgent method. 
+          Agents.CreateAgent populates the name automatically. Format: projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>.
```

```suggestion
          The unique identifier of the agent.
```

We're going to make it so that only the `<Agent ID>` portion is stored here later, so let's remove the other information. Plus, from a Terraform user's perspective the specific methods called don't matter and they never interact with the value outside of import.